src/downloadAzureProject/setupProjectFolder.ts

```diff
@@ -0,0 +1,103 @@
+                const defaultHostName: string | undefined = slotTreeItem?.client.defaultHostName;
+
+                if (!!hostKeys && hostKeys.masterKey && defaultHostName) {
+                    const requestOptions: RequestPrepareOptions = {
+                        method: 'GET',
+                        url: `https://${defaultHostName}/admin/functions/download?includeCsproj=true&includeAppSettings=true`,
+                        headers: { 'x-functions-key':  hostKeys.masterKey }
+                    };
+                    await requestUtils.downloadFile(requestOptions, downloadFilePath);
+                } else {
+                    throw new Error(localize('hostInformationNotFound', 'Failed to get host information for the functionApp.'));
+                }
+            }
+
+            const projectFilePathUri: vscode.Uri = vscode.Uri.joinPath(vsCodeFilePathUri, `${functionAppName}`);
+            const projectFilePath: string = projectFilePathUri.fsPath;
+            const devContainerFolderPathUri: vscode.Uri = vscode.Uri.joinPath(projectFilePathUri, '.devcontainer');
+            if (downloadAppContent) {
+                // tslint:disable-next-line: no-unsafe-any
+                await extract(downloadFilePath, { dir: projectFilePath });
+            }
+            await requestUtils.downloadFile(
+                `https://raw.githubusercontent.com/microsoft/vscode-dev-containers/master/containers/${devContainerName}/.devcontainer/devcontainer.json`,
+                vscode.Uri.joinPath(devContainerFolderPathUri, 'devcontainer.json').fsPath
+            );
+            await requestUtils.downloadFile(
+                `https://raw.githubusercontent.com/microsoft/vscode-dev-containers/master/containers/${devContainerName}/.devcontainer/Dockerfile`,
+                vscode.Uri.joinPath(devContainerFolderPathUri, 'Dockerfile').fsPath
+            );
+            await initProjectForVSCode(context, projectFilePath, getProjectLanguageForLanguage(language));
+            vscode.window.showInformationMessage(localize('restartingVsCodeInfoMessage', 'Restarting VS Code with your function app project'));
+            vscode.commands.executeCommand('vscode.openFolder', vscode.Uri.file(projectFilePath), true);
+        });
+    } catch (err) {
+        throw new Error(localize('failedLocalProjSetupErrorMessage', 'Failed to set up your local project: "{0}".', parseError(err).message));
+    } finally {
+        vscode.workspace.fs.delete(
+            vscode.Uri.file(toBeDeletedFolderPathUri.fsPath),
+            {
+                recursive: true,
+                useTrash: true
+            }
+        );
+    }
+}
+
+function getProjectLanguageForLanguage(language: string): ProjectLanguage {
+    switch (language) {
+        case 'powershell':
+            return ProjectLanguage.PowerShell;
+        case 'node':
+            return ProjectLanguage.TypeScript;
```

Comment:
Shouldn't this be JavaScript? I didn't think the portal supported TypeScript...

Reply:
Sure will update